jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HttpStream.java

```diff
@@ -46,6 +46,22 @@ public interface HttpStream extends Callback
      */
     String getId();
 
+    /**
+     * Signal the intention to read the request content for the purposes of 100-Continues processing as follows: <ul>
+     *     <li>If a {@code 100-Continue} response is expected, then calling this method will send that response and the expectation
+     *     is satisfied. Otherwise this method is a {@code noop}.</li>
+     *     <li>There is an implied call to {@code willRead()} when {@link #demand()} is called.</li>
+     *     <li>If a {@link #read()} is attempted before calling this method and some content is available, then
+     *         any 100-Continues expectation is cancelled.</li>
+     *     <li>If a {@link #send(MetaData.Request, MetaData.Response, boolean, ByteBuffer, Callback)} is done prior to
+     *     a call to {@code willRead()}, then any expectation is cancelled and the connection put into a protocol dependent
+     *     state to not expect the request content to be sent and to ignore any further content sent on the channel</li>
+     * </ul>
+     */
+    default void willRead()
```

Comment:
I think this shouldn't be marked as `default`.

Reply:
ditto